stdlib/public/Concurrency/TaskPrivate.h

```diff
@@ -168,35 +169,96 @@ class TaskFutureWaitAsyncContext : public AsyncContext {
 
 /// The current state of a task's status records.
 class alignas(sizeof(void*) * 2) ActiveTaskStatus {
-  enum : uintptr_t {
+
+  /// 32 bit systems with SWIFT_CONCURRENCY_ENABLE_PRIORITY_ESCALATION=1
+  ///
+  ///          Flags                 Drain Lock             Unused           TaskStatusRecord *
+  /// |----------------------|----------------------|----------------------|-------------------|
+  ///          32 bits                32 bits                32 bits              32 bits
+  ///
+  ///
+  /// 64 bit systems with SWIFT_CONCURRENCY_ENABLE_PRIORITY_ESCALATION=1
+  ///
+  ///         Flags               Drain Lock          TaskStatusRecord *
+  /// |----------------------|-------------------|----------------------|
+  ///          32 bits                32 bits              64 bits
+  ///
+  /// 32 bit systems with SWIFT_CONCURRENCY_ENABLE_PRIORITY_ESCALATION=0
+  ///
+  ///          Flags             TaskStatusRecord *
+  /// |----------------------|----------------------|
+  ///          32 bits                32 bits
+  //
+  /// 64 bit systems with SWIFT_CONCURRENCY_ENABLE_PRIORITY_ESCALATION=0
+  ///
+  ///         Flags                  Unused            TaskStatusRecord *
+  /// |----------------------|----------------------|---------------------|
+  ///         32 bits                 32 bits                64 bits
+  ///
+  /// Note: This layout is intentional since we can then do 64bit atomics on
+  /// just the top 64 bits of atomic state when
+  /// SWIFT_CONCURRENCY_ENABLE_PRIORITY_ESCALATION=1 if we don't need to
+  /// coordinate with the status record pointer.
```

Comment:
Understood, I might tweak your suggestion and omit information about the syscall interface on darwin because that is really more about documentation about dispatch's interfaces and shouldn't live here. Not to mention that it can change in the future.